lib/autoproj/cli/cache.rb

```diff
@@ -1,9 +1,54 @@
+# frozen_string_literal: true
+
 require 'autoproj/cli/inspection_tool'
 require 'autoproj/ops/cache'
 
 module Autoproj
     module CLI
         class Cache < InspectionTool
+            def parse_gem_compile(string)
+                scanner = StringScanner.new(string)
+                name = scanner.scan(/[^\[]+/)
+
+                level = 0
+                artifacts = []
+                artifact_include = nil
+                artifact_name = ''.dup
+                until scanner.eos?
+                    c = scanner.getch
+                    if level == 0
+                        raise ArgumentError, "expected '[' but got '#{c}'" unless c == '['
```

Comment:
How is this possible?

Reply:
Syntax error from the user, e.g. `qtbindings[+lib/2.5/*.so.?]-lib/2.5/*.so]`